crates/fuel-streams-core/src/types.rs

```diff
@@ -49,6 +70,42 @@ macro_rules! generate_byte_type_wrapper {
 
 generate_byte_type_wrapper!(Address, fuel_tx::Address);
 generate_byte_type_wrapper!(Bytes32, fuel_tx::Bytes32);
+generate_byte_type_wrapper!(ContractId, fuel_tx::ContractId);
+generate_byte_type_wrapper!(AssetId, fuel_types::AssetId);
+
+/// Macro to implement conversion from a type to `Bytes32`.
+///
+/// This macro creates an implementation of the `From` trait, allowing for conversion
+/// from the specified type into a `Bytes32` type. It is useful when working with
+/// byte-based types, such as `ContractId`, in the Fuel ecosystem.
+///
+/// The generated implementation allows conversion by dereferencing the input value
+/// and creating a `Bytes32` type from it, making the conversion simple and efficient.
+///
+/// # Usage
+///
+/// impl_from_bytes32!(FromType);
+///
+///
+/// Where `FromType` is the type that you want to be able to convert into a `Bytes32`.
+//
+/// # Notes
+///
+/// The macro assumes that the type being converted can be dereferenced into a byte-based type
+/// compatible with the `Bytes32` structure.
+macro_rules! impl_from_bytes32 {
```

Comment:
Just a thought but - would not it be better to move this macro to the macros crate (fuel-streams-macros) ?

Reply:
Good thinking. However, I think the responsibility of `fuel-streams-macros` is to expose procedural macros that, by design, require a `proc-macro` only crate. 

These declarative macros are enhancements for `fuel-core` types which accurately fit into the responsibility of `fuel-streams-core`.